resources/assets/components/CampaignSingle/index.js

```diff
@@ -20,21 +22,30 @@ class CampaignSingle extends React.Component {
 
     this.api = new RestApiClient;
     this.filterPosts = this.filterPosts.bind(this);
-    this.getPostsByTag = this.getPostsByTag.bind(this);
-    this.getPostsByStatus = this.getPostsByStatus.bind(this);
+    this.getPostsByFilter = this.getPostsByFilter.bind(this);
     this.getPostsByPaginatedLink = this.getPostsByPaginatedLink.bind(this);
   }
 
   // Filter posts based on status or tag(s).
-  filterPosts(filter) {
-    // If the filter is a status, make API call to get posts by status.
-    if (['pending', 'accepted', 'rejected'].includes(filter)) {
-      this.getPostsByStatus(filter, this.props.campaign.id);
-    } else {
-      // If the filter is a tag, make the API call to get posts by tag.
-      this.getPostsByTag(filter, this.props.campaign.id);
+  filterPosts(state) {
```

Comment:
I think you should maybe pass in the filter prop from the state instead of the whole state object. 

So if `state` looked like: 
```
{
  loadingNewPosts: false,
  filters: {
   //.....
  }
}
```
you would pass in just `state.filters` here.